### ISSUE TITLE
CapacitorSQLite.requestPermissions() not working as expected on Android

### ISSUE THREAD

**Issue by accron-1** [capacitor-community/sqlite#48]
**Describe the bug**
`await CapacitorSQLite.requestPermissions()` does not resume execution after storage permissions are granted on Android.

**To Reproduce**
1. Add `await CapacitorSQLite.requestPermissions()` where execution of app starts.
2. Run app on emulator or physical device

**Expected behavior**
The app should halt the execution on `requestPermissions()` and display a permission dialog, and **resume** the execution after the permissions have been granted.
E.g. given the following code:
```
async created() {
    try {
      console.log('0')
      await CapacitorSQLite.requestPermissions();
      console.log('1')
    }
    catch(e) {
      console.log(e);
    }
  }
```
I would expect the to see `0` and `1` logged after permission has been granted.

However, what happens is that only `0` is logged and the permission dialog is displayed, but nothing after `requestPermissions()` is executed. See screenshot below.

**Screenshots**
![Logs](https://user-images.githubusercontent.com/14213307/97333663-15760700-187c-11eb-8eb9-50765011ac0c.png)

**Smartphone (please complete the following information):**
- Device: Emulator and physical device
- OS: Android
- Version: 10

**Comment by accron-1** [comment]
Can someone confirm that it is supposed to work as I describe under *Expected behavior*?

**Comment by jepiqueau** [comment]
I see your request, I was busy on something else and did not consider it as urgent. I will have a look tomorrow if I can reproduce it and come back

**Comment by jepiqueau** [comment]
@accron-1 
I tested it both with Ionic/Angular and Ionic/React and it works fine 
for Ionic/Angular i defined a sqlite.service.ts see(`https://github.com/jepiqueau/angular-sqlite-app-starter.git`)
```
import { Injectable } from '@angular/core';

import { Plugins } from '@capacitor/core';
import '@capacitor-community/sqlite';
const { CapacitorSQLite, Device } = Plugins;

@Injectable({
  providedIn: 'root'
})
export class SQLiteService {
  sqlite: any;
  isService: boolean = false;
  platform: string;
  constructor() {
  }
  /**
   * Plugin Initialization
   */
  async initializePlugin(): Promise<void> {
    const info = await Device.getInfo();
    this.platform = info.platform;
    console.log("*** platform " + this.platform)
    this.sqlite = CapacitorSQLite;
    this.isService = true;

    if(this.platform === "android") {
      console.log("%%% in androidPremissions platform " 
      + this.platform + "%%%");
      try {
        console.log("%%% before requesting permissions %%%")
        await this.sqlite.requestPermissions();
        console.log("%%% after requesting permissions %%%")
      } catch (e) {
        console.log("Error requesting permissions " + e);
        this.isService = false;
      }
    }
    return;
  }
  /**
   * Get Echo 
   * @param value string 
   */
  async getEcho(value:string): Promise<any> {
    if (this.isService) {
      return await this.sqlite.echo({value:"Hello from JEEP"});
    } else {
      return Promise.resolve("");
    }
  }
  /**
   * Open a Database
   ...
   ...
}
```


```
I/Capacitor/Console: File: http://localhost/default~tab2-tab2-module~tab3-tab3-module-es2015.js - Line 1623 - Msg: %%% before requesting permissions %%%
V/Capacitor/Plugin: To native (Capacitor plugin): callbackId: 55996610, pluginId: CapacitorSQLite, methodName: requestPermissions
V/Capacitor: callback: 55996610, pluginId: CapacitorSQLite, methodName: requestPermissions, methodData: {}
D/Capacitor: App paused
I/Choreographer: Skipped 33 frames!  The application may be doing too much work on its main thread.
D/Capacitor/App: Firing change: true
V/Capacitor/App: Notifying listeners for event appStateChange
D/Capacitor/App: No listeners found for event appStateChange
D/Capacitor: App resumed
V/Capacitor/Network: Notifying listeners for event networkStatusChange
D/Capacitor/Network: No listeners found for event networkStatusChange
I/Capacitor/Console: File: http://localhost/default~tab2-tab2-module~tab3-tab3-module-es2015.js - Line 1625 - Msg: %%% after requesting permissions %%%
I/Capacitor/Console: File: http://localhost/tab2-tab2-module-es2015.js - Line 232 - Msg: ****** entering run tests
```
So for me the issue can be closed

**Comment by accron-1** [comment]
Thank you for looking into it.

I originally reported it in the Ionic forum [here](https://forum.ionicframework.com/t/how-handle-storage-permissions/198518/9).
Since then another user in that thread has reported the same issue, I'm not the only person with that problem it seems.

FWIW I'm using the plugin with a Vue based app (Quasar). Maybe the lifecycles are handled differently in the different languages/platforms?

-------
Another thing that I didn't report in my original report above is that, after I grant the storage permissions, that line of code
`await CapacitorSQLite.requestPermissions()`
literally crashes the app every time I open it. I have to comment it out in order to run my app.

So I think it's still very much an open issue.

**Comment by jepiqueau** [comment]
@accron-1 I am going to provide soon a `vue-sqlite-hook` and a Vue3 apps which will demonstrate how to use it.

**Comment by accron-1** [comment]
So it's completely expected that the plugin doesn't work in Vue, until you release that hook?

**Comment by jepiqueau** [comment]
@accron-1 which version of the plugin are you using?  I will not close the issue till i demonstrate it works in Vue. You will have to wait few days. I put this as my top priority. Thanks any how for your comments..

**Comment by jepiqueau** [comment]
@accron-1 Are you using a pure Vue3 js app or a Ionic/vue ts app ? so i can fix my priority

**Comment by accron-1** [comment]
It's the stable v. 2.4.0

It's a mobile Android app based on [Quasar](https://quasar.dev), which is Vue. For now Quasar supports Vue2 only.
However, the [Quasar mobile apps](https://quasar.dev/quasar-cli/developing-capacitor-apps/introduction) are in turn completely based on Ionic's Capacitor, so if you get it to work on Ionic Capacitor-Vue, I hope it will work on Quasar as well.

The only possible caveat is that Ionic Vue is Vue3, while Quasar is currently Vue2, though I think they are working on a Vue3 upgrade.

Does that answer your question?

**Comment by jepiqueau** [comment]
@accron-1 You should use the 2.4.2-8 which has got a lot of fixes. my version number depends on @capacitor/core version. So i will go to 2.4.2 as soon as Ionic will release 2.4.3 and after this i will go to 2.4.3-1 for my fixes and so on
I do not know Quasar so here i cannot help you too much. so try 2.4.2-8 with Quasar and Vue2 and tell me if it works

**Comment by jepiqueau** [comment]
@accron-1 the requestPermission for Android has been implemented in 2.4.1-1

**Comment by accron-1** [comment]
Thank you for the quick reply. I updated to but 2.4.2-8, but unfortunately little has changed. The app still crashes on first startup, and on subsequent runs.

My real init code looks like this:
```
async created() {
    try {
      console.log('0')
      await CapacitorSQLite.requestPermissions();
      console.log('1')
      await db.open();
      console.log('2')
      await this.loadSettings();
      console.log('3')
      this.handleAuthStateChanged();
    }
    catch(e) {
      console.log(e);
    }
  }
```

This logs the following:

![log1](https://user-images.githubusercontent.com/14213307/97865674-dc6de480-1d0a-11eb-9e1a-099ade126b19.png)

The only difference is that it now also logs `1`, which it didn't do in the previous version, however it seems to crash when it tries to open the db.

If I comment out
`await CapacitorSQLite.requestPermissions()`
everything works as expected:

![log2](https://user-images.githubusercontent.com/14213307/97865816-1808ae80-1d0b-11eb-926d-cb469f056b47.png)

The Android Studio logs are as follows: [AndroidStudioPluginCrashLog.txt](https://github.com/capacitor-community/sqlite/files/5474824/AndroidStudioPluginCrashLog.txt)

Hope that helps.
Feel free to ask me if you have questions.

**Comment by jepiqueau** [comment]
@accron-1  Well i do not know what is `await db.open();` means and on what `db` point to.
the correct way to open a database is:

```
        const r = await capacitorSQLite.open({database: mDatabase});
        if(!r.result) {
            console.lof(r.message);
        }
        console.log('2')
       ...
       ...
```

**Comment by jepiqueau** [comment]
@accron-1  where `mDatabase` is the name of your database (type string)

**Comment by accron-1** [comment]
`db` is my **own** db object that executes the sql against your api, i.e. it's just a wrapper around your plugin `CapacitorSQLite`.

Inside my `db.open()` I do a bunch of stuff, including calling your api `await db.open({database: "db_name"});`, so that's not the issue, or else it wouldn't work when commenting out the line `await CapacitorSQLite.requestPermissions()` either.

`loadSettings()` also calls the db for loading the app settings, which you can see in all the `CapacitorSQLite.query` logs in the second screenshot, and which works fine.

The point of posting the complete code was to demonstrate difference in execution when `requestPermissions()` runs and doesn't: running it crashes the app.

**Comment by jepiqueau** [comment]
@accron-1 a bit strange try to move the requestPermissions inside your db.open it seems that there is a `capacitorSQLite.query`  before the `capacitorSQLite.open`

**Comment by jepiqueau** [comment]
@accron-1 i got now everything working for Vue3 see `https://github.com/jepiqueau/vue-sqlite-app.git` using `https://github.com/jepiqueau/vue-sqlite-hook.git`

**Comment by accron-1** [comment]
I had a db call in another child component, inside the main component, that did the db call: I missed the fact that the child components will be created and executed regardless of what happens inside the parent component, so that's why there was a db call before the db was opened, while the parent component was waiting for the `CapacitorSQLite.requestPermissions()` input. My bad, I'm very sorry.

It's all been corrected now and I can confirm that the latest version of the plugin does indeed work, even with Quasar and Vue2. Please close this issue.

Apologies for the distraction, and thank you for your work on a great and crucial plugin!

**Comment by jepiqueau** [comment]
@accron-1 I am please to hear that it works now. Do not apology, it is part of software development, the error could have been from my side, this is why i was pretty concerned. It force me to change my priority and propose quickly a `vue-sqlite-hook` which was certainly impatiently waited by the community as overnight 46 developers downloaded it. Good luck with your app. By curiosity what Quasar is doing?

**Comment by jepiqueau** [comment]
@accron-1 i close the issue

**Comment by accron-1** [comment]
Thank you, my app is already in the Play store.

The number of downloads you've had proves how important your plugin is. That's why I think the focus of your work should be of that kind, rather than things like ORM support.

Quasar is a [Vue based framework](https://quasar.dev/introduction-to-quasar#What-is-Quasar%3F), but unlike Ionic you can also write web and electron apps with Quasar, not only mobile apps.

When I started developing my app, Ionic/Vue hadn't been released yet, so after a search for Vue based alternatives I came across Quasar. I didn't know about Quasar either before then, but it's an awesome framework with a bunch of premade components (expand the section '*Vue Components*' on the left side side to see a list of all of them). I don't understand why it isn't more known than it is. I would use Quasar for any new mobile apps from now on, even when Ionic/Vue is available. I highly recommend you try it if you're interested.

**Comment by accron-1** [comment]
(I don't want to open an issue for a similar question, so I'll ask here)

I updated to v.3.2.0, but it's not clear how to do the import in an Ionic Capacitor app.
Currently I do `import { CapacitorSQLite } from '@capacitor-community/sqlite'`. Is that correct?
I'm asking because I get the following error in Android Studio on startup:
`java.lang.ClassCastException: class com.getcapacitor.community.database.sqlite.CapacitorSQLite cannot be cast to com.getcapacitor.Plugin`

Also, what is the correct way to do `CapacitorSQLite.requestPermissions()` now? I checked the latest docs but couldn't find anything.
